File: language-testing/guess_number.lua
#! /usr/bin/env lua

math.randomseed(os.time())
local solution = math.floor(math.random(1, 100))

print [[
Guess-a-number game!
I'm thinking of a number between 1 and 100,
you have 5 tries to guess it correctly.
What's your guess?
]]

local guess, guess_num = 0, 0
while 1 do
  guess_num = guess_num + 1
  print(string.format("Guess %d: ", guess_num))
  guess = io.read("*number")

  if guess < solution then
    print "Too low! Try again!"
  elseif guess > solution then
    print "Too high! Try again!"
  elseif guess == solution then
    print "Correct! You won!"
    os.exit()
  end

  if guess_num > 4 then
    print(string.format("Haha, I won! The number was %d!", solution))
    os.exit()
  end
end

--[[ TAIL INFO:
Name: Guess Number
Language: Python3
State: Done

Play guess-a-number game



Example ./guess_number.py
]]

File: language-testing/guess_number.py
#! /usr/bin/env python3

from random import randint

print("Guess-a-number game!",
      "I am thinking of a number between 1 and 100",
      "What's your guess?",
      sep="\n")

target = randint(1, 100)
number = 0
current = 0

while 1:

    number += 1
    current = int(input("Guess %d: " % number))

    if current == target:
        print("Correct! You won!")
        break
    elif number == 5:
        print("Haha, I won! The number was %d" % target)
        break
    elif current < target: 
        print("Too low! Guess again!")
    elif current > target: 
        print("Too high! Guess again!")



"""
Name: Guess Number
Language: Python3
State: Done

Play guess-a-number game



Example ./guess_number.py
"""
